packages/patternfly-4/react-charts/src/components/ChartDonutUtilization/ChartDonutThreshold.js

```diff
@@ -0,0 +1,258 @@
+   * height of the chart in number of pixels, but instead define an aspect ratio for the chart. The exact number of
+   * pixels will depend on the size of the container the chart is rendered into.
+   *
+   * Note: innerRadius may need to be set when using this property.
+   */
+  width: PropTypes.number,
+  /**
+   * The x prop specifies how to access the X value of each data point.
+   * If given as a function, it will be run on each data point, and returned value will be used.
+   * If given as an integer, it will be used as an array index for array-type data points.
+   * If given as a string, it will be used as a property key for object-type data points.
+   * If given as an array of strings, or a string containing dots or brackets,
+   * it will be used as a nested object property path (for details see Lodash docs for _.get).
+   * If `null` or `undefined`, the data value will be used as is (identity function/pass-through).
+   * @example 0, 'x', 'x.value.nested.1.thing', 'x[2].also.nested', null, d => Math.sin(d)
+   */
+  x: PropTypes.string,
+  /**
+   * The y prop specifies how to access the Y value of each data point.
+   * If given as a function, it will be run on each data point, and returned value will be used.
+   * If given as an integer, it will be used as an array index for array-type data points.
+   * If given as a string, it will be used as a property key for object-type data points.
+   * If given as an array of strings, or a string containing dots or brackets,
+   * it will be used as a nested object property path (for details see Lodash docs for _.get).
+   * If `null` or `undefined`, the data value will be used as is (identity function/pass-through).
+   * @example 0, 'y', 'y.value.nested.1.thing', 'y[2].also.nested', null, d => Math.sin(d)
+   */
+  y: PropTypes.string
+};
+
+const ChartDonutThreshold = ({
+  children,
+  data = [],
+  donutOrientation = 'left',
+  legendOrientation,
+  standalone = true,
+  themeColor,
+  themeVariant,
+  x,
+  y,
+
+  // destructure last
+  theme = getDonutThresholdStaticTheme(themeColor, themeVariant),
+  donutHeight = theme.pie.height,
+  donutWidth = theme.pie.width,
+  height = theme.pie.height,
+  innerRadius = ((donutHeight || donutWidth) - 34) / 2,
+  width = theme.pie.width,
+  ...rest
+}) => {
+  // Returns computed data representing pie chart slices
+  const getComputedData = () => {
+    const computedData = [];
+    let prevYVal = 0;
+    data.forEach(_data => {
+      const datum = [{ ..._data }];
+      const xVal = getDatumXVal(datum);
+      const yVal = getDatumYVal(datum);
+      computedData.push({ x: xVal, y: yVal ? Math.abs(yVal - prevYVal) : 0 });
+      prevYVal = yVal;
+    });
+    computedData.push({ y: prevYVal ? Math.abs(100 - prevYVal) : 0 });
+    return computedData;
+  };
+
+  // Returns a data point value via the x accessor
+  const getDatumXVal = datum => {
+    const result = map(datum, property(x || 'x'));
```

Comment:
We may be able to use the `formatData` helper function from Victory.

Reply:
Replaced with Victory's formatData function.